internal/commitremap/commitremap.go

```diff
@@ -1,129 +1,158 @@
 package commitremap
 
 import (
+	"bufio"
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"log"
 	"os"
 	"path/filepath"
 	"strings"
+	"sync"
 )
 
-// Struct to represent a single entry in the commit map
-type CommitMapEntry struct {
-	Old string
-	New string
+type File struct {
+	FilePath string
+	Prefix   string
 }
 
 // Parses the file and returns a map of old commit hashes to new commit hashes
-func ParseCommitMap(filePath string) (*[]CommitMapEntry, error) {
-	commitMap := []CommitMapEntry{}
-
+func ParseCommitMap(filePath string) (*map[string]string, error) {
+	commitMap := make(map[string]string)
 	// Read the commit-map file
 	content, err := os.ReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
-	// Split the file content into lines
-	lines := strings.Split(string(content), "\n")
-
-	// Iterate over the lines and parse the old and new commit hashes
-	for _, line := range lines {
-		if strings.TrimSpace(line) == "" {
+	buf := bytes.NewBuffer(content)
+	scanner := bufio.NewScanner(buf)
+	for scanner.Scan() {
+		line := scanner.Text()
+		// Skip adding the header to the map
+		if line == "old                                      new" {
 			continue
 		}
-
-		fields := strings.Fields(line)
+		fields := strings.Split(line, " ")
 		if len(fields) != 2 {
 			return nil, fmt.Errorf("invalid line: %s", line)
 		}
-
-		commitMap = append(commitMap, CommitMapEntry{
-			Old: fields[0],
-			New: fields[1],
-		})
+		commitMap[fields[0]] = fields[1]
 	}
 	return &commitMap, nil
 }
 
-func ProcessFiles(archiveLocation string, prefixes []string, commitMap *[]CommitMapEntry) error {
-
-	for _, prefix := range prefixes {
-		// Get a list of all files that match the pattern
-		files, err := filepath.Glob(filepath.Join(archiveLocation, prefix+"_*.json"))
+func ProcessFiles(archiveLocation string, prefixes []string,
+	commitMap *map[string]string, workers int) error {
+	workerCount := 10
+	fileChannel := make(chan File, workerCount)
+	fileProcessWg := sync.WaitGroup{}
+	filesToProcess := getAllFilesToProcess(prefixes, archiveLocation)
+	totalFiles := len(filesToProcess)
+	processedFiles := make(chan File, totalFiles)
+	processedFilesCount := 0
+	// go routine to print out the progress of the processed files. It also
+	// writes the processed files to a log file
+	fmt.Printf("Processed %d/%d files\n", processedFilesCount, totalFiles)
+	go func() {
+		f, err := os.OpenFile("processed_files.log", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 		if err != nil {
-			log.Fatalf("Error getting files: %v", err)
+			log.Fatalf("Error opening processed files log: %v", err)
 		}
-
-		// Process each file
-		for _, file := range files {
-			log.Println("Processing file:", file)
-
-			err := updateMetadataFile(file, commitMap)
-			if err != nil {
-				return fmt.Errorf("Error updating metadata file: %v; %v", file, err)
+		defer f.Close()
+		for file := range processedFiles {
+			fmt.Printf("\033[1A\033[K")
```

Comment:
O.o can you explain this one ? 😆

Reply:
Yeah good point, I will add a comment there 😃

Reply:
Comment makes sense of the ASCII jargon. I'm not sure of a better way to delete the previous line in the terminal.... 😭